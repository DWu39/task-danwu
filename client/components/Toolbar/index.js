import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import Button from '../Button';
import theme from '../../style/theme';

const Container = styled.div`
align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 24px 0;
`;

const StyledHeading = styled.p`
  color: ${theme.navy};
  font-size: 28px;
  margin: 0;
  text-transform: capitalize;
`;

const ButtonGroup = styled.div`
  > button:first-child {
    margin-right: 12px;
  }
`;

const Toolbar = ({
  edited,
  newTask,
  saveTasks
}) => (
  <Container>
    <StyledHeading>tasks</StyledHeading>
    <ButtonGroup>
      <Button primary={false} onClick={newTask}>add task</Button>
      <Button disabled={!edited} onClick={saveTasks}>save</Button>
    </ButtonGroup>
  </Container>
);

Toolbar.propTypes = {
  edited: PropTypes.bool,
  newTask: PropTypes.func,
  saveTasks: PropTypes.func
};

Toolbar.defaultProps = {
  edited: false,
  newTask: null,
  saveTasks: null
}

export default Toolbar;
