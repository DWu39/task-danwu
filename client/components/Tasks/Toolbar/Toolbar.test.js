import React from 'react';
import { mount, shallow } from 'enzyme';

import Toolbar from './index';

describe('Toolbar', () => {
  it('should render correctly', () => {
    const output = shallow(<Toolbar/>);
    expect(output).toMatchSnapshot();
  });

  it('should add task on \'Add Task\' click', () => {
    const mockNewTaskHandler = jest.fn();
    const output = mount(<Toolbar newTask={mockNewTaskHandler}/>);
    const button = output.find('button').first();
    button.simulate('click');
    expect(mockNewTaskHandler).toHaveBeenCalledTimes(1);
  });

  it('should NOT save on \'Save\' click when state is NOT edited', () => {
    const saveTasks = jest.fn();
    const output = mount(<Toolbar saveTasks={saveTasks}/>);
    const button = output.find('button').at(1);
    button.simulate('click');
    expect(saveTasks).not.toHaveBeenCalled();
  });

  it('should save on \'Save\' click when state is edited', () => {
    const saveTasks = jest.fn();
    const output = mount(<Toolbar edited saveTasks={saveTasks}/>);
    const button = output.find('button').at(1);
    button.simulate('click');
    expect(saveTasks).toHaveBeenCalledTimes(1);
  });
});
