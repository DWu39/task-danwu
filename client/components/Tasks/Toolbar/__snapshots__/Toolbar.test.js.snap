// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Toolbar
    edited={false}
    newTask={null}
    saveTasks={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.p>
          tasks
        </styled.p>,
        <styled.div>
          <Button
            className={null}
            disabled={false}
            onClick={null}
            primary={false}
          >
            add task
          </Button>
          <Button
            className={null}
            disabled={true}
            onClick={null}
            primary={true}
          >
            save
          </Button>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "tasks",
        },
        "ref": null,
        "rendered": "tasks",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              className={null}
              disabled={false}
              onClick={null}
              primary={false}
            >
              add task
            </Button>,
            <Button
              className={null}
              disabled={true}
              onClick={null}
              primary={true}
            >
              save
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "add task",
              "className": null,
              "disabled": false,
              "onClick": null,
              "primary": false,
            },
            "ref": null,
            "rendered": "add task",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "save",
              "className": null,
              "disabled": true,
              "onClick": null,
              "primary": true,
            },
            "ref": null,
            "rendered": "save",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.p>
            tasks
          </styled.p>,
          <styled.div>
            <Button
              className={null}
              disabled={false}
              onClick={null}
              primary={false}
            >
              add task
            </Button>
            <Button
              className={null}
              disabled={true}
              onClick={null}
              primary={true}
            >
              save
            </Button>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "tasks",
          },
          "ref": null,
          "rendered": "tasks",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                className={null}
                disabled={false}
                onClick={null}
                primary={false}
              >
                add task
              </Button>,
              <Button
                className={null}
                disabled={true}
                onClick={null}
                primary={true}
              >
                save
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "add task",
                "className": null,
                "disabled": false,
                "onClick": null,
                "primary": false,
              },
              "ref": null,
              "rendered": "add task",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "save",
                "className": null,
                "disabled": true,
                "onClick": null,
                "primary": true,
              },
              "ref": null,
              "rendered": "save",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
